backend/infrahub/core/account.py

```diff
@@ -145,24 +145,93 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
         WITH account, r1 as r
         WHERE r.status = "active"
         WITH account
-        MATCH group_path = (account)-[]->(:Relationship {name: "group_member"})
-            <-[]-(:%(account_group_node)s)
-            -[]->(:Relationship {name: "role__accountgroups"})
-            <-[]-(:%(account_role_node)s)
-            -[]->(:Relationship {name: "role__permissions"})
-            <-[]-(object_permission:%(object_permission_node)s)
-            -[:HAS_ATTRIBUTE]->(:Attribute {name: "branch"})
-            -[:HAS_VALUE]->(object_permission_branch:AttributeValue)
+        CALL {
+            WITH account
+            MATCH (account)-[r1:IS_RELATED]->(:Relationship {name: "group_member"})<-[r2:IS_RELATED]-(account_group:%(account_group_node)s)
+            WHERE all(r IN [r1, r2] WHERE (%(branch_filter)s))
+            WITH account_group, r1, r2, (r1.status = "active" AND r2.status = "active") AS is_active
+            ORDER BY account_group.uuid, r2.branch_level DESC, r2.from DESC, r1.branch_level DESC, r1.from DESC
+            WITH account_group, head(collect(is_active)) as latest_is_active
+            WHERE latest_is_active = TRUE
+            RETURN account_group
+        }
+        WITH account_group
```

Comment:
each of these subqueries does the same things. in pseudocode
- get all the possible paths to the next type of node (`CoreAccountGroup` in this one)
- order all of these paths by `account_group`, branch_depth (b/c we prefer deeper branch paths), and from time
- for each `account_group`, get the latest, deepest path from `account`
- for each `account_group`, only keep it if its latest, deepest path has all active edges